### ISSUE TITLE
List<String> being generated as List<Enum>

### ISSUE THREAD

**Issue by panshul007** [swagger-api/swagger-codegen#2100]
I updated swagger-codegen from version 2.1.2 to 2.1.5
Using Java 8
Using the provided JavaClientCodegen.java file
I have the following specification: 
```json
{
  "features": {
    "description": "contains the list of features to the store (e.g. Restaurant, Fish, ...)",
    "type": "array",
    "items": {
      "enum": [
        "Restaurant",
        "Fish",
        "ParkingSpot",
        "Wine",
        "Meat"
      ],
      "type": "string"
    }
  }
}
```

and it generates:
```java
 private List<String> features = new ArrayList<FeaturesEnum>();
```

Please help. What am I doing wrong.

**Comment by fehguy** [comment]
That looks like a bug, I don't think you're doing anything wrong.  We'll try to address it in short order.

**Comment by panshul007** [comment]
:+1:

**Comment by xhh** [comment]
Tried the above property spec against latest master and it works for me:

```java
private List<FeaturesEnum> features = new ArrayList<FeaturesEnum>();
```

@panshul007 how did you run the generation?

**Comment by lopesmcc** [comment]
Same happening with me. 
Generating for language 'spring-mvc' with library 'j8-async', using the latest swagger-codegen-maven-plugin (2.1.5).

**Comment by lopesmcc** [comment]
Java client generation seems to be working correctly in master, so it must have been fixed already.
SpringMVC still had this issue and I created a PR with the fix ( #2249 ).

**Comment by wing328** [comment]
I believe this issue has been addressed. Please reply if that's not the case.
